miden-crypto/src/aead/xchacha/mod.rs

```diff
@@ -179,6 +222,38 @@ impl SecretKey {
             .decrypt(&nonce.inner, payload)
             .map_err(|_| EncryptionError::FailedOperation)
     }
+
+    // ELEMENT DECRYPTION
+    // --------------------------------------------------------------------------------------------
+
+    /// Decrypts the provided encrypted data using this secret key.
+    ///
+    /// Note that if the original data was encrypted as bytes (e.g., using [Self::encrypt_bytes()]
+    /// method), the decryption will still succeed but an additional step will need to be taken to
+    /// convert the returned field elements into the original bytestring.
+    pub fn decrypt_elements(
+        &self,
+        encrypted_data: &EncryptedData,
+    ) -> Result<Vec<Felt>, EncryptionError> {
+        self.decrypt_elements_with_associated_data(encrypted_data, &[])
+    }
+
+    /// Decrypts the provided encrypted data, given some associated data, using this secret key.
+    ///
+    /// Note that if the original data was encrypted as bytes (e.g., using [Self::encrypt_bytes()]
+    /// method), the decryption will still succeed but an additional step will need to be taken to
+    /// convert the returned field elements into the original bytestring.
+    pub fn decrypt_elements_with_associated_data(
+        &self,
+        encrypted_data: &EncryptedData,
+        associated_data: &[Felt],
+    ) -> Result<Vec<Felt>, EncryptionError> {
+        let ad_bytes = elements_to_bytes_unchecked(associated_data);
+
+        let plaintext_bytes = self.decrypt_bytes_with_associated_data(encrypted_data, &ad_bytes)?;
+
+        Ok(bytes_to_elements_unchecked(&plaintext_bytes))
```

Comment:
Wouldn't this function fail on maliciously serialized data? For example, if we try to decrypt data that was just bytes, `plaintext_bytes` would not contain valid filed elements, and then this call here would panic.

I think during the conversion of bytes to elements, we should check two things:
- The length of `plaintext_bytes` is divisible by 8. Otherwise, the last field element would be truncated.
- All 8-byte sequences contain valid filed elements.

Reply:
That's a good point, updated to include your suggestions